src/components/editor/MonacoEditor.vue

```diff
@@ -0,0 +1,67 @@
+<template>
+  <div class="monaco-editor"
+    data-cy="monaco-editor"
+  >
+    <div id="container"></div>
+  </div>
+</template>
+
+<script setup>
+import {
+  onMounted,
+  onUpdated,
+  nextTick,
+} from 'vue';
+
+const monaco = require('monaco-editor');
+
+// Need to be here, otherwise OnUpdated is not call
+defineProps({
+  viewType: String,
+});
+
+let editor;
+
+/**
+ * Setup monaco editor.
+ */
+function createEditor() {
+  const container = document.getElementById('container');
+  editor = monaco.editor.create(container, {
+    value: '',
+    language: 'javascript',
+  });
+}
+
+/**
+ * Update default editor layout size.
+ * Needed when viewType is firstly set to 'model' and ModelizerTextView is hidden,
+ * meaning 'container' height and width are set to 0.
+ */
+function updateEditorLayout() {
+  const container = document.getElementById('container');
+  editor.layout({
+    height: container.offsetHeight,
+    width: container.offsetWidth,
+  });
+}
+
+onMounted(() => createEditor());
```

Comment:
I guess (when there'll be some text in) it'll take a not insignificant amount of time to load otherwise.

At some point, the editor instantiation should probably be cached. Only keep one instance of it in memory.

Reply:
I'm agree with that, but we will reoganize the editor when we will integrate plugin management.